fe/fe-core/src/main/java/com/starrocks/sql/analyzer/FunctionAnalyzer.java

```diff
@@ -99,6 +100,12 @@ public static void analyze(FunctionCallExpr functionCallExpr) {
                     "map_apply's lambda function can not be null");
             functionCallExpr.setType(functionCallExpr.getChild(0).getChild(0).getType());
         }
+
+        if (fnName.getFunction().equals(FunctionSet.STR_TO_MAP)) {
+            Type retType = new MapType(Type.VARCHAR, Type.VARCHAR);
+            functionCallExpr.setType(retType);
+            functionCallExpr.getFn().setRetType(retType);
+        }
```

Comment:
should resovle it in PolymorphicFunctionAnalyzer?

Reply:
as it is speical that all its parameters are not Polymorphic.

Reply:
so, if this is a sepcial type function, why not register it by special type? like:
```
    [170004, 'str_to_map', 'MAP<VARCHAR, VARCHAR>', ['MAP<VARCHAR, VARCHAR>', 'VARCHAR', 'VARCHAR'], 'nullptr'],
```

be may we need support this way?

Reply:
added map_varchar_varchar